node -v v0.12.2 mac os，install the grpc official example,an error occurred

i cross-reference this article:http://www.grpc.io/docs/installation/node.html

then run this code:

$ git clone https://github.com/grpc/grpc.git

$ cd examples/node
$ npm install
then happen the follow error:

whwdeMacBook-Pro:node whw$ npm install

/

> grpc@0.11.1 install /Users/whw/grpc

> node-gyp rebuild



usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]

[-e pattern] [-f file] [--binary-files=value] [--color=when]

[--context[=num]] [--directories=action] [--label] [--line-buffered]

[--null] [pattern] [file ...]

gyp: Call to 'node -v | grep -oP "(?<=v)(\d+)(?=\.\d+\.\d+)"' returned exit status 2. while trying to load binding.gyp

gyp ERR! configure error 

gyp ERR! stack Error: `gyp` failed with exit code: 1

gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:343:16)

gyp ERR! stack     at ChildProcess.emit (events.js:110:17)

gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)

gyp ERR! System Darwin 14.5.0

gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"

gyp ERR! cwd /Users/whw/grpc

gyp ERR! node -v v0.12.2

gyp ERR! node-gyp -v v1.0.3

gyp ERR! not ok 



npm ERR! Darwin 14.5.0

npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"

npm ERR! node v0.12.2

npm ERR! npm  v2.7.4

npm ERR! code ELIFECYCLE

npm ERR! grpc@0.11.1 install: `node-gyp rebuild`

npm ERR! Exit status 1

npm ERR! 

npm ERR! Failed at the grpc@0.11.1 install script 'node-gyp rebuild'.

npm ERR! This is most likely a problem with the grpc package,

npm ERR! not with npm itself.

npm ERR! Tell the author that this fails on your system:

npm ERR!     node-gyp rebuild

npm ERR! You can get their info via:

npm ERR!     npm owner ls grpc

npm ERR! There is likely additional logging output above.



npm ERR! Please include the following file with any support request:

npm ERR!     /Users/whw/grpc/examples/node/npm-debug.log

--- a/binding.gyp
+++ b/binding.gyp
@@ -46,10 +46,11 @@
       # io.js always reports versions >0 and always exports ALPN symbols.
       # Therefore, Node's major version will be truthy if and only if it
       # supports ALPN. The output of "node -v" is v[major].[minor].[patch],
-      # like "v4.1.1" in a recent version. We use grep to extract just the
-      # major version. "4", would be the output for the example.
+      # like "v4.1.1" in a recent version. We use cut to split by period and
+      # take the first field (resulting in "v[major]"), then use cut again
+      # to take all but the first character, removing the "v".
     'defines': [
-      'TSI_OPENSSL_ALPN_SUPPORT=<!(node -v | grep -oP "(?<=v)(\d+)(?=\.\d+\.\d+)")'
+      'TSI_OPENSSL_ALPN_SUPPORT=<!(node --version | cut -d. -f1 | cut -c2-)'
     ],
     'include_dirs': [
       '.',

--- a/templates/binding.gyp.template
+++ b/templates/binding.gyp.template
@@ -48,10 +48,11 @@
         # io.js always reports versions >0 and always exports ALPN symbols.
         # Therefore, Node's major version will be truthy if and only if it
         # supports ALPN. The output of "node -v" is v[major].[minor].[patch],
-        # like "v4.1.1" in a recent version. We use grep to extract just the
-        # major version. "4", would be the output for the example.
+        # like "v4.1.1" in a recent version. We use cut to split by period and
+        # take the first field (resulting in "v[major]"), then use cut again
+        # to take all but the first character, removing the "v".
       'defines': [
-        'TSI_OPENSSL_ALPN_SUPPORT=<!(node -v | grep -oP "(?<=v)(\d+)(?=\.\d+\.\d+)")'
+        'TSI_OPENSSL_ALPN_SUPPORT=<!(node --version | cut -d. -f1 | cut -c2-)'
       ],
       'include_dirs': [
         '.',

